cmake/pika_setup_cuda.cmake

```diff
@@ -151,7 +151,7 @@ function(pika_add_nvhpc_cuda_flags source)
   get_source_file_property(source_compile_flags ${source} COMPILE_FLAGS)
   string(REGEX REPLACE "^NOTFOUND$" "" source_compile_flags ${source_compile_flags})
   set_source_files_properties(
-    ${source} PROPERTIES COMPILE_FLAGS "${source_compile_flags} -x cu ${source_gpu_cc_flags}"
-                         LANGUAGE CXX
+    ${source} PROPERTIES COMPILE_FLAGS "${source_compile_flags} -x cu ${source_gpu_cc_flags} \
+      --diag_suppress cuda_compile" LANGUAGE CXX
```

Comment:
Can you move this to https://github.com/pika-org/pika/blob/4a94194ecb53fa74cf1c8b6b5f8fb7b7aa556aad/cmake/pika_setup_cuda.cmake#L24? I realize now that we may need to guard that flag with a generator expression for the compiler that is used (so that if consumers compile with e.g. GCC they don't get these flags).

Reply:
Never mind the second part. I think it's accidentally handled by us recreating the `pika_internal::cuda` target when someone does `find_package(pika)`.

Reply:
Ah, you're right! Sorry for the noise, this seems like a good place to keep it, thanks.